### PR TITLE
GitHub workflows

### PR DESCRIPTION
**Pull request by musangowope**
**Proposed workflow:**
- Once you have merged your feature into master, run `lerna version --conventional-commits` in the terminal.
- You will be prompted with semvar options for updating the packages. (`major`, `minor`, `patch` etc.)
- Select an option and an automatic tagged commit will occur and the changelog for the packages will be updated.
- Create a new release [here](https://github.com/laws-africa/la-web-components/releases/new). Use the tag that was pushed
  for your release.
- Upon creating a release, a new version of the packages will be published on `npm`.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Publish
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
```
[the remark this answers is not in the document]
Comment (longhotsummer) [reply .github/workflows/publish.yml:6]: See https://github.com/laws-africa/la-web-components/blob/master/.github/workflows/publish.yml

**Comment by musangowope** [comment]
The unit test passes on my local machine but fails github actions. Looking into this.

**Comment by musangowope** [comment]
@longhotsummer 
So `law-widgets-styles` visual tests pass if we have a workflow that runs on `macos-latest`
```
name: Unit Tests

on:
  push:
    branches:
      - "*"

jobs:
  run_tests:
    runs-on: macos-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2.1.5
        with:
          node-version: 14.x
      - run: npm ci
      - run: npx lerna bootstrap
      - run: npx lerna run test
```

But ideally, we want to run on ubuntu because it saves on costs
https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Install dependencies and build 🔧
+        run: | # Install and link dependencies
+          npm ci
+          npx lerna bootstrap
+
+      - name: Version and publish
+        id: graduateRelease
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor}}@users.noreply.github.com"
```
Comment (longhotsummer) [inline .github/workflows/publish.yml:31]: These can be removed, I think they are unused

**Review by longhotsummer — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (musangowope) [reply .github/workflows/publish.yml:31]: `user.name and user.email` is required used for 
```
npx lerna publish from-package -y
```

**Pull request merged**